File: remove_bg.py
from PIL import Image
import sys

def remove_white_bg(filename):
    img = Image.open(filename)
    img = img.convert("RGBA")
    datas = img.getdata()
    newData = []
    for item in datas:
        if item[0] > 230 and item[1] > 230 and item[2] > 230:
            newData.append((255, 255, 255, 0))
        else:
            newData.append(item)
    img.putdata(newData)
    img.save(filename, "PNG")

remove_white_bg('/Users/sahid/Projects/inspiro-fashion/public/assets/floating_tshirt.png')
remove_white_bg('/Users/sahid/Projects/inspiro-fashion/public/assets/floating_jacket.png')
remove_white_bg('/Users/sahid/Projects/inspiro-fashion/public/assets/floating_sneaker.png')
remove_white_bg('/Users/sahid/Projects/inspiro-fashion/public/assets/floating_cap.png')

File: remove_bg.js
const { Jimp } = require("jimp");

async function removeWhiteBg(filename) {
    const image = await Jimp.read(filename);
    image.scan(0, 0, image.bitmap.width, image.bitmap.height, function(x, y, idx) {
        const red = this.bitmap.data[idx + 0];
        const green = this.bitmap.data[idx + 1];
        const blue = this.bitmap.data[idx + 2];
        if (red > 230 && green > 230 && blue > 230) {
            this.bitmap.data[idx + 3] = 0; // Alpha 0
        }
    });
    await image.write(filename);
}

async function processAll() {
    await removeWhiteBg('/Users/sahid/Projects/inspiro-fashion/public/assets/floating_tshirt.png');
    await removeWhiteBg('/Users/sahid/Projects/inspiro-fashion/public/assets/floating_jacket.png');
    await removeWhiteBg('/Users/sahid/Projects/inspiro-fashion/public/assets/floating_sneaker.png');
    await removeWhiteBg('/Users/sahid/Projects/inspiro-fashion/public/assets/floating_cap.png');
}

processAll().catch(console.error);
